tests/core/io/test_json_native.h

```diff
@@ -188,9 +176,7 @@ TEST_CASE("[JSON][Native] Conversion between native and JSON formats") {
 	CHECK(decode(arr2_repr).get_construct_string() == "[1, null, 9]");
 	ERR_PRINT_ON;
 
-	TypedArray<Resource> res_arr;
-	res_arr.push_back(res);
-
+	TypedArray<Resource> res_arr{ res };
```

Comment:
This change seems to make the test fail for some reason.

Reply:
Should be solved by adding a dedicated constructor to `TypedArray`, this probably goes via `Array` now which then is converted incorrectly

Reply:
I only added the initializer list for the specialized TypedArrays (`MAKE_TYPED_ARRAY`), so it doesn't work with objects.
This goes through the `TypedArray(const Variant &p_variant)` non-macro constructor and I think it tries to convert the object into an array so it is left empty.
I didn't add it since I was having issues with it, but I was probably doing something wrong.

Adding an initializer list constructor to the non-macro TypedArray should fix it, or I can open a PR for it.
```gdscript
_FORCE_INLINE_ TypedArray(std::initializer_list<Variant> p_init) : TypedArray(Array(p_init)) {}
```